aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+  }
+
+  @Override
+  public void close() throws IOException {
+    glue.close();
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    String dbName = IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace());
+    String tableName = tableIdentifier.name();
+    return new GlueTableOperations(conf, glue, catalogId, dbName, tableName);
+  }
+
+  /**
+   * Use the same format as Hive for default warehouse location
+   * @param tableIdentifier table id
+   * @return default warehouse path
+   */
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    // check if value is set in database
+    GetDatabaseResponse response = glue.getDatabase(GetDatabaseRequest.builder()
+        .name(IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace()))
+        .build());
+    String dbLocationUri = response.database().locationUri();
+    if (dbLocationUri != null) {
+      return String.format("%s/%s", dbLocationUri, tableIdentifier.name());
+    }
+
+    return String.format(
+        "%s/%s.db/%s",
+        warehousePath,
+        IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace()),
+        tableIdentifier.name());
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    namespaceExists(namespace);
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<TableIdentifier> results = new ArrayList<>();
+    while (true) {
+      GetTablesResponse response = glue.getTables(GetTablesRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(IcebergToGlueConverter.toDatabaseName(namespace))
+          .nextToken(nextToken)
+          .build());
+      if (response.hasTableList()) {
+        results.addAll(response.tableList().stream()
+            .map(GlueToIcebergConverter::toTableId)
+            .collect(Collectors.toList()));
+        nextToken = response.nextToken();
+      }
+      if (nextToken == null) {
+        break;
+      }
+    }
+    return results;
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    try {
+      dropTable(newTableOps(identifier), identifier, purge);
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  private TableMetadata dropTable(TableOperations ops, TableIdentifier identifier, boolean purge) {
+    TableMetadata lastMetadata = null;
+    if (ops.current() != null) {
+      lastMetadata = ops.current();
+    }
+
+    glue.deleteTable(DeleteTableRequest.builder()
+        .catalogId(catalogId)
+        .databaseName(IcebergToGlueConverter.toDatabaseName(identifier.namespace()))
+        .name(identifier.name())
+        .build());
+
+    if (purge && lastMetadata != null) {
+      CatalogUtil.dropTableData(ops.io(), lastMetadata);
+    }
+
+    LOG.info("Dropped table: {}", identifier);
```

Comment:
Nit: Maybe add an info log above when the table is dropped from the catalog and then one here about whether or not the table was purged?